File: lecture-1-C/pset1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

// call helper function
int get_height(void);
void print_grid(int size);

// MAIN FUNCTION
int main(void)
{
    int n = get_height();
    print_grid(n);
}

// Helper function
int get_height(void)
{
    // initialize int height
    int height;
    // do... while loop in case height < 1 || > 8
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    return height;
}

// Helper function
void print_grid(int size)
{
    // nested for loop
    for (int i = 0; i < size; i++)
    {
        // nested for loop to make pyramid face to right side
        for (int j = i + 1; j < size; j++)
        {
            // print space
            printf(" ");
        }
        // nested for loop for actual pyramid
        for (int hash = size + i + 1; hash > size; hash--)
        {
            // print #
            printf("#");
        }
        // print in new line
        printf("\n");
    }
}

// test: $ check50 cs50/problems/2023/x/mario/less
// style: $ style50 mario.c
// submit: $ submit50 cs50/problems/2023/x/mario/less

File: lecture-6-Python/pset6/sentimental-mario-less/mario.py
# Mario
# Write a program that recreates the half-pyramid using # for blocks
# prompt user with get_int for a positive integer between 1-8 inclusive
# If user fails to provide a positive integer no greater than 8 we will re-prompt user
# generate the half pyramid

from cs50 import get_int


def main():
    """Main function that takes in user input for height and prints out desired pyramid based on user height"""
    height = get_height()
    print_grid(height)


def get_height():
    """Helper function that calculates the height"""
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break
    return height


def print_grid(size):
    """Helper function to print the pyramid"""
    for i in range(0, size, 1):
        for j in range(0, size, 1):
            if (i + j < size - 1):
                print(" ", end="")
            else:
                print("#", end="")
        print()


main() 

# test - $ check50 cs50/problems/2023/x/sentimental/mario/less
# style - $ style50 mario.py
# submit - $ submit50 cs50/problems/2023/x/sentimental/mario/less
